runtime/onert/backend/trix/DevContext.cc

```diff
@@ -137,60 +139,64 @@ void DevContext::requestRun(ModelID model_id, input_buffers *input_bufs, tensors
-      }
-      assert(input_bufs->bufs[0].size % batch_size == 0);
-      const uint64_t in_batch_offset = input_bufs->bufs[0].size / batch_size;
-      setBufferByBatch(input_bufs->bufs[0], batch_num, in_batch_offset, &in_batch_buffers->bufs[0]);
-
-      auto out_batch_buffers = &out_buffers_vec.at(batch_num);
-      out_batch_buffers->num_buffers = output_bufs->num_buffers;
-      if (output_bufs->bufs[0].size % batch_size != 0)
-      {
-        throw std::runtime_error(
-          "Invalid batch size. batch size :" + std::to_string(batch_size) +
-          ", output tensor size : " + std::to_string(output_bufs->bufs[0].size));
-      }
-      const uint64_t out_batch_offset = output_bufs->bufs[0].size / batch_size;
-      setBufferByBatch(output_bufs->bufs[0], batch_num, out_batch_offset,
-                       &out_batch_buffers->bufs[0]);
-
-      // Set data info of inputs/outputs for each batch
-      // TODO Support multiple inputs/outputs
-      in_infos.at(batch_num) = *in_info;
-      out_infos.at(batch_num) = *out_info;
-
       // Enqueue jobs
       // The in_info and out_info are always the same even if they are divided by batch, so they are
       // used as they are.
       auto future = _batch_thread_pool->enqueueJob(
-        [this](uint32_t dev_num, ModelID model_id, input_buffers *input_bufs,
-               tensors_data_info *in_info, output_buffers *output_bufs, tensors_data_info *out_info,
-               uint32_t batch_num) -> int32_t {
-          // dev_num is the same as the thread number in _batch_thread_pool
-          this->runOneBatch(dev_num, model_id, input_bufs, in_info, output_bufs, out_info);
+        [batch_size, in_info, out_info,
+         this](uint32_t dev_num, ModelID model_id, const input_buffers *input_bufs,
+               const output_buffers *output_bufs, uint32_t batch_num) -> int32_t {
+          // Set buffers of inputs/outputs for each batch
+          // TODO Support multiple inputs/outputs
+          input_buffers in_batch_buffers;
+          in_batch_buffers.num_buffers = input_bufs->num_buffers;
+          const uint64_t in_batch_offset = input_bufs->bufs[0].size / batch_size;
+          setBufferByBatch(input_bufs->bufs[0], batch_num, in_batch_offset,
+                           &in_batch_buffers.bufs[0]);
+
+          output_buffers out_batch_buffers;
+          out_batch_buffers.num_buffers = output_bufs->num_buffers;
+          const uint64_t out_batch_offset = output_bufs->bufs[0].size / batch_size;
+          setBufferByBatch(output_bufs->bufs[0], batch_num, out_batch_offset,
+                           &out_batch_buffers.bufs[0]);
+
+          try
+          {
+            // dev_num is the same as the thread number in _batch_thread_pool
+            this->runOneBatch(dev_num, model_id, &in_batch_buffers, in_info, &out_batch_buffers,
+                              out_info);
+          }
+          catch (...)
+          {
+            _eptr = std::current_exception();
```

Comment:
(Q) Are you trying to see only the last exception that occurred?

Reply:
I only saw an explanation that it was a current exception. I think that the current exception is the same as the last exception is is a try-catch block.